functions/src/winloss.ts

```diff
@@ -0,0 +1,125 @@
+import { firestore } from "firebase-admin";
+import * as functions from "firebase-functions";
+import { db } from "./app";
+import { getPrivatePlayerCollection, lobbyCollection } from "./firestore-collections";
+import { isLobbyRequest } from "./firebase-functions-types";
+import { Lobby, PrivatePlayer } from "./firestore-types/lobby";
+
+// this function will likely be obsolete once timers are introduced
+export const lobbyReturn = functions.https.onCall(async (data: unknown, context): Promise<void> => {
+  if (context.auth === undefined) {
+    throw new functions.https.HttpsError("permission-denied", "Not Signed In");
+  }
+
+  // validate code
+  if (!isLobbyRequest(data)) {
+    throw new functions.https.HttpsError("invalid-argument", "Invalid lobby code!");
+  }
+
+  // get lobby doc
+  const lobby = await lobbyCollection.doc(data.code).get();
+  if (lobby.exists === false) {
+    throw new functions.https.HttpsError("not-found", "Lobby doesn't exist!");
+  }
+
+  const { uids, state } = lobby.data() as Lobby;
+  // check if host
+  if (context.auth.uid !== uids[0]) {
+    throw new functions.https.HttpsError("permission-denied", "Not the host of the game!");
+  }
+
+  // check game state
+  if (state !== "END") {
+    throw new functions.https.HttpsError("invalid-argument", "Wrong game state!");
+  }
+
+  await resetLobby(lobby.ref);
+  await deletePrivatePlayers(lobby.ref, getPrivatePlayerCollection(lobby.ref));
```

Comment:
each of these functions get the lobby doc wich causes a total of 3 reads. instead lobby should be passed into these functions

Reply:
`getPrivatePlayerCollection` requires a lobby ref document, so would it be fine to leave that as is?

Reply:
> `getPrivatePlayerCollection` requires a lobby ref document, so would it be fine to leave that as is?

I'm not 100% sure what you're asking.

I think you're asking if it is ok to pass lobby and keep `getPrivatePlayerCollection(lobby.ref)` the way it is

and if that's what you are asking then yes

Reply:
yup, that was all I was wondering. Should be good then.